src/selectors/Entities/nameNote.js

```diff
@@ -11,3 +13,53 @@ export const selectNameNoteIsValid = state => {
   const { isValid = false } = nameNoteState ?? {};
   return isValid;
 };
+
+// Selects most recent PCD entered into the system at the time of vaccination
+// (i.e. closest before or at the same time as the vaccination entry)
+export const selectClosestPCDToVaccination = (patient, vaccinationEntryDate) => {
+  const [pcdEvent] = UIDatabase.objects('PCDEvents');
+  if (!pcdEvent) return null;
+
+  const { id: pcdEventID } = pcdEvent;
+  const nameNotes = patient?.nameNotes ?? [];
+
+  if (!nameNotes.length) return null;
+
+  // Check first for a nameNote entered before the vaccination date
+  const filtered = nameNotes.filter(
+    ({ patientEventID, entryDate }) =>
+      patientEventID === pcdEventID && moment(vaccinationEntryDate) >= moment(entryDate)
+  );
+
+  if (!filtered.length) return null;
+
+  const sorted = filtered.sort(
+    ({ entryDate: entryDateA }, { entryDate: entryDateB }) =>
+      moment(entryDateB).valueOf() - moment(entryDateA).valueOf()
+  );
+
+  const [mostRecentPCD] = sorted;
+  return mostRecentPCD;
+};
+
+export const selectMostRecentPCD = patient => {
```

Comment:
this and the method above are doing almost the same thing; what about creating a `selectMostRecentNameNote` which accepted a filter parameter? then both methods could use the new one

Reply:
Ooh, yes they did end up being almost identical. Thanks :) https://github.com/openmsupply/mobile/pull/4668/commits/2b825ed120fd78bbc2e88375da5553f9a359335e